src/analysis/retail/shaman/enhancement/modules/talents/HotHand.tsx

```diff
@@ -55,75 +83,120 @@ interface HotHandProc extends SpellCast {
  *
  */
 class HotHand extends MajorCooldown<HotHandProc> {
-  description(): ReactNode {
-    throw new Error('Method not implemented.');
-  }
-  explainPerformance(cast: HotHandProc): SpellUse {
-    throw new Error('Method not implemented.');
-  }
   static dependencies = {
     ...MajorCooldown.dependencies,
     spellUsable: SpellUsable,
     haste: Haste,
+    abilities: Abilities,
   };
   protected spellUsable!: SpellUsable;
   protected haste!: Haste;
+  protected abilities!: Abilities;
+
+  activeWindow: HotHandProc | null = null;
+  globalCooldownEnds: number = 0;
 
   protected hotHand!: HotHandRank;
   protected buffedLavaLashDamage: number = 0;
   protected hotHandActive: Intervals = new Intervals();
   protected buffedCasts: number = 0;
 
   constructor(options: Options) {
-    super({ spell: TALENTS_SHAMAN.LAVA_LASH_TALENT }, options);
+    super({ spell: TALENTS_SHAMAN.HOT_HAND_TALENT }, options);
     this.active = this.selectedCombatant.hasTalent(TALENTS_SHAMAN.HOT_HAND_TALENT);
     if (!this.active) {
       return;
     }
 
     this.hotHand = HOT_HAND[this.selectedCombatant.getTalentRank(TALENTS_SHAMAN.HOT_HAND_TALENT)];
 
-    this.addEventListener(
-      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.HOT_HAND_BUFF),
-      this.applyHotHand,
-    );
+    const abilities = options.abilities as Abilities;
+    abilities.add({
+      spell: SPELLS.HOT_HAND_BUFF.id,
+      category: SPELL_CATEGORY.COOLDOWNS,
+      gcd: null,
+      enabled: true,
+    });
```

Comment:
this is a buff, it shouldn't be added to `Abilities`. it isn't something that you cast and a bunch of stuff like cast efficiency doesn't make sense for it.

Reply:
Done